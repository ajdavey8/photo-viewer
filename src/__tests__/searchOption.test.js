import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import SearchOption from '../components/searchOption';

describe(SearchOption, () => {
  const file = { "file": "1 Alan Berryhill 2018-04-17 person, outdoors, rain.jpg"};
  const component = shallow(<SearchOption result={file} />);

  it('renders correctly and matches snapshot', () => {
    const tree = renderer.create(<SearchOption result={file} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the photographers name', () => {
    expect(component.find('p').text()).toContain("Alan Berryhill");
  })
});
