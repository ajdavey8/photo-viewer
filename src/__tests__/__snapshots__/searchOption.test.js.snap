// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`({ result }) => {

  const photographer = file => {
    const name = file.split(" ");
    return name[1] + " " + name[2];
  };

  return (
    _react2.default.createElement('div', { __source: { fileName: _jsxFileName, lineNumber: 13 }, __self: undefined },
      _react2.default.createElement('div', { className: 'thumb', __source: { fileName: _jsxFileName, lineNumber: 14 }, __self: undefined },
        _react2.default.createElement(_photo2.default, { image: result, __source: { fileName: _jsxFileName, lineNumber: 15 }, __self: undefined })),

      _react2.default.createElement('div', { className: 'optionname', __source: { fileName: _jsxFileName, lineNumber: 17 }, __self: undefined },
        _react2.default.createElement('p', { __source: { fileName: _jsxFileName, lineNumber: 18 }, __self: undefined }, ' ', photographer(result.file), ' '))));



} renders correctly and matches snapshot 1`] = `
<div>
  <div
    className="thumb"
  >
    <img
      alt="1 Alan Berryhill 2018-04-17 person, outdoors, rain.jpg"
      src="1 Alan Berryhill 2018-04-17 person, outdoors, rain.jpg"
    />
  </div>
  <div
    className="optionname"
  >
    <p>
       
      Alan Berryhill
       
    </p>
  </div>
</div>
`;
